ignite/cmd/chain.go

```diff
@@ -60,13 +72,79 @@ chain.
 		Args:    cobra.ExactArgs(1),
 	}
 
-	c.AddCommand(
-		NewChainServe(),
-		NewChainBuild(),
-		NewChainInit(),
-		NewChainFaucet(),
-		NewChainSimulate(),
-	)
+	c.AddCommand(addConfigMigrationVerifier(NewChainServe()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainBuild()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainInit()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainFaucet()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainSimulate()))
 
 	return c
 }
+
+func addConfigMigrationVerifier(cmd *cobra.Command) *cobra.Command {
+	cmd.Flags().AddFlagSet(flagSetConfig())
+	cmd.Flags().AddFlagSet(flagSetYes())
+
+	preRunFun := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) (err error) {
+		if preRunFun != nil {
+			if err = preRunFun(cmd, args); err != nil {
+				return err
+			}
+		}
+
+		configPath := getConfig(cmd)
+		if configPath == "" {
+			appPath := flagGetPath(cmd)
+
+			if configPath, err = chainconfig.LocateDefault(appPath); err != nil {
+				return err
+			}
+		}
+
+		rawCfg, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return err
+		}
+
+		version, err := chainconfig.ReadConfigVersion(bytes.NewReader(rawCfg))
+		if err != nil {
+			return err
+		}
+
+		// Config files with older versions must be migrated to the latest before executing the command
+		if version != chainconfig.LatestVersion {
```

Comment:
At this stage, what about stopping immediately if we detect the config file has uncommited changes ? We just output a message asking the user to commit his config to proceed the migration.

Reply:
Yes! It makes total sense to avoid the migration when the repository state is not clean.

Reply:
Done in 53afcd90fe7da9cc5be913f077a058dce85e17d3